django_project/frontend/src/containers/MainPage/Upload/Step1.tsx

```diff
@@ -52,34 +53,54 @@ export default function Step1(props: Step1Interface) {
             setPropertyValidation(_error_validation)
             return
         }
+        // validate property name
+        let _checkNameData: any = {
+            'name': property.name.trim()
+        }
         if (property.id) {
-            // if it has property.id, then this is to update property
-            let _data:PropertyInterface = {
-                ...property
-            }
-            setSavingProperty(true)
-            postData(`${PROPERTY_UPDATE_INFORMATION_URL}`, _data).then(
-                response => {
-                    setSavingProperty(false)
-                    // trigger to next step
-                    props.onSave({...property})
-                }
-              ).catch(error => {
-                setSavingProperty(false)
-                console.log('error ', error)
-                alert('Error saving property...')
-              })
-        } else {
-            props.onSave({...property})
+            _checkNameData['id'] = property.id
         }
+        postData(`${PROPERTY_CHECK_AVAILABLE_NAME_URL}`, _checkNameData).then(
+            response => {
+                let _isAvailable = response.data['available']
+                if (_isAvailable) {
+                    if (property.id) {
+                        // if it has property.id, then this is to update property
+                        let _data:PropertyInterface = {
+                            ...property
+                        }
+                        setSavingProperty(true)
+                        postData(`${PROPERTY_UPDATE_INFORMATION_URL}`, _data).then(
```

Comment:
@danangmassandy could we extract this into a new function instead?

Reply:
updated Dimas,,